ciao-controller/configuration.go

```diff
@@ -0,0 +1,265 @@
+	if validConfigValue(newConf.Value, newConf.Element) == false {
+		return errors.New("invalid Element")
+	}
+	pyld, err = c.client.ssntpClient().ClusterConfiguration()
+	if err != nil {
+		return err
+	}
+	err = testNewConfig(newConf, token, tenantID, pyld)
+	if err != nil {
+		return err
+	}
+	err = updateConfPayload(newConf, &pyld)
+	if err != nil {
+		return err
+	}
+	newConfPyld, err := yaml.Marshal(pyld)
+	if err != nil {
+		return err
+	}
+	c.client.ssntpClient().SendCommand(ssntp.CONFIGURE, newConfPyld)
+	return nil
+}
+
+func (c *controller) ShowClusterConfig() (string, error) {
+	pyld, err := c.client.ssntpClient().ClusterConfiguration()
+	if err != nil {
+		return "", err
+	}
+	// no need to share sensitive information
+	pyld.Configure.Controller.IdentityPassword = "*****"
+	pyld.Configure.Controller.AdminPassword = "*****"
+	pyld.Configure.Controller.AdminSSHKey = "*****"
+	s, err := yaml.Marshal(pyld)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s", string(s)), nil
+}
+
+func updateConfPayload(confReq types.ConfigRequest, pyld *payloads.Configure) error {
+	badValueErr := fmt.Errorf("Unable to set '%s' value for '%s'", confReq.Value, confReq.Element)
+	switch confReq.Element {
+	case "scheduler.storage_uri":
+		pyld.Configure.Scheduler.ConfigStorageURI = confReq.Value
+	case "controller.compute_port":
+		newVal, err := strconv.Atoi(confReq.Value)
+		if err != nil {
+			return badValueErr
+		}
+		pyld.Configure.Controller.ComputePort = newVal
+	case "launcher.disk_limit":
+		newVal, err := strconv.ParseBool(confReq.Value)
+		if err != nil {
+			return badValueErr
+		}
+		pyld.Configure.Launcher.DiskLimit = newVal
+	case "launcher.mem_limit":
+		newVal, err := strconv.ParseBool(confReq.Value)
+		if err != nil {
+			return badValueErr
+		}
+		pyld.Configure.Launcher.MemoryLimit = newVal
+	case "identity_service.type":
+		if confReq.Element != "keystone" {
+			return badValueErr
+		}
+	case "identity_service.url":
+		pyld.Configure.IdentityService.URL = confReq.Value
+	}
+	return nil
+}
+
+func testRequest(method string, url string, headers map[string]string, body io.ReadCloser, expRetCodes []int) error {
+	req, err := http.NewRequest(method, url, nil)
+	if err != nil {
+		return fmt.Errorf("Unable to create request for new values check: %v", err)
+	}
+	for k, v := range headers {
+		req.Header.Add(k, v)
+
+	}
+	if body != nil {
+		req.Body = body
+	}
+
+	tlsConfig := &tls.Config{}
+
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+
+	client := &http.Client{Transport: transport}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("Could not send HTTP test request %s", err)
+	}
+	for _, retCode := range expRetCodes {
+		if retCode == resp.StatusCode {
+			return nil
+		}
+	}
+	return fmt.Errorf("Bad status code received: %v (expecting %v)", resp.StatusCode, expRetCodes)
+}
+
+func testNewConfig(newConf types.ConfigRequest, token string, tenantID string, pyld payloads.Configure) error {
```

Comment:
Yes that would be my recommendation.  @kaccardi do you have a preference on what do to here?